tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/UpgradePlanCorePlugin.java

```diff
@@ -152,46 +156,71 @@ private void _initOfflineOutlines() throws Exception {
 			return;
 		}
 
-		while (entryUrls.hasMoreElements()) {
-			URL fileURL = FileLocator.toFileURL(entryUrls.nextElement());
+		try {
+			while (entryUrls.hasMoreElements()) {
+				URL fileURL = FileLocator.toFileURL(entryUrls.nextElement());
 
-			File outlineFile = new File(fileURL.getFile());
+				File outlineFile = new File(fileURL.getFile());
 
-			String contentZipMD5 = "";
+				String outlineFilename = outlineFile.getName();
 
-			switch (outlineFile.getName()) {
-				case "code-upgrade.zip":
-					contentZipMD5 = CODE_UPGRADE_ZIP_MD5;
+				Job outlineJob = new Job("Unzip upgrade doc: " + outlineFilename) {
 
-					break;
+					@Override
+					protected IStatus run(IProgressMonitor monitor) {
+						try {
+							String contentZipMD5 = "";
 
-				case "database-upgrade.zip":
-					contentZipMD5 = DATABASE_UPGRADE_ZIP_MD5;
+							switch (outlineFile.getName()) {
+								case "code-upgrade.zip":
+									contentZipMD5 = CODE_UPGRADE_ZIP_MD5;
 
-					break;
-			}
+									break;
 
-			String outlineFilename = outlineFile.getName();
+								case "database-upgrade.zip":
+									contentZipMD5 = DATABASE_UPGRADE_ZIP_MD5;
 
-			String outlineFilenameWithoutEx = outlineFilename.split("\\.")[0];
+									break;
+							}
 
-			String storedMD5 = preferencesService.getString(UpgradePlanCorePlugin.ID, contentZipMD5, "", null);
+							String outlineFilenameWithoutEx = outlineFilename.split("\\.")[0];
 
-			String updateMD5 = _computeMD5(outlineFile);
+							String storedMD5 = preferencesService.getString(
+								UpgradePlanCorePlugin.ID, contentZipMD5, "", null);
 
-			IPath offlineDocDirPath = offlineOutlinePath.append(outlineFilenameWithoutEx);
+							String updateMD5 = _computeMD5(outlineFile);
 
-			if (!updateMD5.equals(storedMD5) || FileUtil.notExists(offlineDocDirPath)) {
-				FileUtil.deleteDir(offlineDocDirPath.toFile(), true);
+							IPath offlineDocDirPath = offlineOutlinePath.append(outlineFilenameWithoutEx);
 
-				ZipUtil.unzip(outlineFile, offlineDocDirPath.toFile());
+							if (!updateMD5.equals(storedMD5) || FileUtil.notExists(offlineDocDirPath)) {
+								FileUtil.deleteDir(offlineDocDirPath.toFile(), true);
 
-				_prefstore.put(contentZipMD5, updateMD5);
+								ZipUtil.unzip(outlineFile, offlineDocDirPath.toFile());
 
-				_prefstore.flush();
-			}
+								_prefstore.put(contentZipMD5, updateMD5);
+
+								_prefstore.flush();
+							}
+
+							_offlineOutlines.add(
+								new UpgradePlanOutline(outlineFilenameWithoutEx, offlineDocDirPath.toOSString()));
+						}
+						catch (Exception e) {
+						}
+
+						return Status.OK_STATUS;
+					}
 
-			_offlineOutlines.add(new UpgradePlanOutline(outlineFilenameWithoutEx, offlineDocDirPath.toOSString()));
+				};
+
+				outlineJob.setProperty(ILiferayProjectProvider.LIFERAY_PROJECT_JOB, new Object());
+
+				outlineJob.setSystem(true);
+
+				outlineJob.schedule();
```

Comment:
hey @simonjhy   Also, can you add a rule here and make sure that this job has finished before we open the "new upgrade plan..." dialog?  So that if the download taking a long time, the wizard wont start until it finishes.

Reply:
This will not spend long time because this job just unzip the offline doc from upgrade plan core plugin and it don't need to download anything.